server/lib/analytics.js

```diff
@@ -8,7 +8,16 @@ const segmentArticle = require('../lib/article/segment');
 const DEBUG = false;
 const BARRIERTYPE = 'trial';
 
-const fixBooleans = data => data.replace('"AUTHDATA(data.access)"', 'AUTHDATA(data.access)');
+const fixAuthData = data => {
+	const authDataString = '"AUTHDATA(access)"';
+
+	// When `access` is `false`, amp-analytics for some reason outputs an empty string.
+	if(authDataString === '""') {
```

Comment:
shouldn't this be `data ===`?

Reply:
No, since as far as I can tell, this file gets parsed at runtime with the `AUTHDATA(access)` being dynamically added in once the auth endpoint returns.

Also, `data` in this case is the entire object getting sent to spoor, but the same stands for `data.context.amp_auth_access` which is where the problem lies. Let me perhaps make it more explicitly about that property.

Reply:
i don't understand how, running on the server, `authDataString` will ever get replaced? i thought amp-analytics did the replacement when sending the request?